### PR TITLE
[MRG] ENH: add verbose param to load_bad_channels, log info

### PR DESCRIPTION
**Pull request by sappelhoff**
I was just using this function and thought that it would be good to have it log about the updates.

I also turned a `warnings.warn` into a `logger.warn` ... however that meant I had to remove a `    with pytest.warns(RuntimeWarning, match='1 bad channel'):` from the test. Is there a way to keep this `with pytest.warns` with the `logger.info`?

Also, should `load_bad_channels` maybe return `self` (i.e., the raw instance)? Similar to `set_annotations` and many other functions?

### CODE REVIEW

**Comment by larsoner** [comment]
We almost never want to use warnings.warn or logger.warning, but instead should use our internal mne.utils.warn. It does stack level magic and also does not warn if verbose level is 'error'

File: mne/io/base.py [hunk 1]
```diff
@@ -1624,34 +1625,39 @@ def load_bad_channels(self, bad_file=None, force=False):
...
                 if not force:
-                    raise ValueError('Bad channels from:\n%s\n not found '
-                                     'in:\n%s' % (bad_file,
-                                                  self.filenames[0]))
+                    raise ValueError(msg)
                 else:
-                    warn('%d bad channels from:\n%s\nnot found in:\n%s'
-                         % (count_diff, bad_file, self.filenames[0]))
-            self.info['bads'] = names_there
+                    logger.warning(msg)
```
Comment (larsoner) [inline mne/io/base.py:1654]: This should be reverted, it was already using our internal mne.utils.warn. Why did you need to change this?

Comment (sappelhoff) [reply mne/io/base.py]: You are right, this was an error on my side ... I assumed this was `warnings.warn` ... but it was `mne.utils.warn`

File: mne/io/fiff/tests/test_raw_fiff.py [hunk 2]
```diff
@@ -574,8 +574,9 @@ def test_load_bad_channels(tmp_path):
 
     # Test forcing the bad case
     with pytest.warns(RuntimeWarning, match='1 bad channel'):
-        raw.load_bad_channels(bad_file_wrong, force=True)
-        # write it out, read it in, and check
+        raw.load_bad_channels(bad_file_wrong, force=True, verbose=None)
```
Comment (larsoner) [inline mne/io/fiff/tests/test_raw_fiff.py:577]: verbose=None here is a noop, you shouldn't need to add this

Comment (sappelhoff) [reply mne/io/fiff/tests/test_raw_fiff.py:577]: I figured it'd be a good smoke test to make sure it accepts that argument?

Comment (larsoner) [reply mne/io/fiff/tests/test_raw_fiff.py:577]: If you really want to check functionally, you can make an identical call with verbose='error' without the warns context manager.

I don't think we need to start adding tests for verbose=None, we use it many places and I wouldn't want to see a bunch of verbose=One's added to our tests

Comment (sappelhoff) [reply mne/io/fiff/tests/test_raw_fiff.py]: okay, I think you are right, I removed that in 04f2e07

**Comment by larsoner** [comment]
Thanks @sappelhoff !
